.github/workflows/github-ci.yml

```diff
@@ -0,0 +1,106 @@
+name: Sub-IoT-Stack CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  run-unit-tests:
+    name: Run Unit Tests
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Build stack natively
+      run: |
+        mkdir build && cd build
+        platform="NATIVE"
+        cmake ../stack/ -DPLATFORM=$platform -DCMAKE_TOOLCHAIN_FILE="../stack/cmake/toolchains/gcc.cmake" -DBUILD_UNIT_TESTS=y -DFRAMEWORK_CONSOLE_ENABLED=n -DTEST_AES=y -DTEST_FEC=y -DTEST_ALP=y -DTEST_FIFO=y -DMODULE_D7AP=n -DMODULE_ALP_SERIAL_INTERFACE_ENABLED=n
+        make -j
+
+    - name: Run unit tests
+      run: |
+        ./build/tests/aes/test_aes &> results_aes.txt
+        ./build/tests/alp/test_alp &> results_alp.txt
+        ./build/tests/fec/test_fec &> results_fec.txt
+        ./build/tests/fifo/test_fifo &> results_fifo.txt
+
+    - name: Upload the results
+      uses: actions/upload-artifact@v2
+      with:
+        name: results
+        path: |
+          results_aes.txt
+          results_alp.txt
+          results_fec.txt
+          results_fifo.txt
+
+    - name: Handle results
+      run: |
+        if ! grep -q 'AES all unit tests OK !' "results_aes.txt"; then exit 1;  fi
+        if ! grep -q 'Unit-tests for ALP completed' "results_alp.txt"; then exit 1;  fi
+        if ! grep -q 'Input was decoded successfully' "results_fec.txt"; then exit 1;  fi
+        if ! grep -q 'All FIFO tests passed!' "results_fifo.txt"; then exit 1;  fi   
+
+  run-testsuite:
+    name: Run Testsuite
+    runs-on: self-hosted
```

Comment:
is it actually possible to run different steps on different runners? for example: building the images could be done on a docker runner but flashing and testing has to run on self-hosted